lms/djangoapps/certificates/models.py

```diff
@@ -878,20 +889,56 @@ def is_enabled_for_course(cls, course_key):
         except cls.DoesNotExist:
             return False
         else:
-            return latest.enabled
+            return latest.can_self_generate
 
     @classmethod
-    def set_enabled_for_course(cls, course_key, is_enabled):
+    def set_self_generation_for_course(cls, course_key, enable):
         """Enable or disable self-generated certificates for a course.
 
         Arguments:
             course_key (CourseKey): The identifier for the course.
-            is_enabled (boolean): Whether to enable or disable self-generated certificates.
+            enable (boolean): Whether to enable or disable self-generated certificates.
+
+        """
+        defaults = {
+            'enabled': enable,
+            'can_self_generate': enable
+        }
+        CertificateGenerationCourseSetting.objects.update_or_create(
+            course_key=course_key,
+            defaults=defaults
+        )
+
+    @classmethod
+    def is_language_specific_templates_enabled_for_course(cls, course_key):
+        """Check whether certificates generated for a course should use language specific templates.
+
+        Arguments:
+            course_key (CourseKey): The identifier for the course.
+
+        Returns:
+            boolean
+
+        """
+        try:
+            latest = cls.objects.filter(course_key=course_key).latest()
```

Comment:
Why do we use `filter` + `latest` instead of just `get`? Can there be more than one row for a given `course_key`?

Reply:
I believe there used to be possibility for more than one row for a given course key, but I also updated the setter to update_or_create instead of just always create, so this can probably change to get.

Reply:
We need to be careful that there aren't actually duplicate entries in the database before switching to `get`. If there's a possibility of there being duplicate rows, I think we still need to use `filter` + `latest`.